### PR TITLE
C2C-78: Add database and file backup cronjobs

### PR DESCRIPTION
**Pull request by enyachoke**

### CODE REVIEW

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+          - name: {{ .Release.Name }}-backup-services-storage
+            persistentVolumeClaim:
+              claimName: {{ .Release.Name }}-backup-services-pvc
+
+---
+
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Release.Name }}-firestore-backup
```
Comment (rbuisson) [inline src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:113]: `firestore`!
->`filestore`

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            persistentVolumeClaim:
+              claimName: {{ .Release.Name }}-backup-services-pvc
+
+
+---
+
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Release.Name }}-postres-backup
```
Comment (rbuisson) [inline src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:59]: `{{ .Release.Name }}-postres-backup` seems like it should be named `{{ .Release.Name }}-odoo-db-backup`

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 3]
```diff
@@ -0,0 +1,254 @@
+{{ if .Values.apps }}{{ if .Values.apps.backup_services }} {{ if .Values.apps.backup_services.enabled }}
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Release.Name }}-mysql-backup
```
Comment (rbuisson) [inline src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:5]: `{{ .Release.Name }}-mysql-backup` seems like it should be named `{{ .Release.Name }}-openmrs-db-backup`

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+spec:
+  successfulJobsHistoryLimit: 2
+  failedJobsHistoryLimit: 2
+  schedule: {{ .Values.apps.backup_services.schedule | quote }}
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: {{ .Release.Name }}-firestore-backup
```
Comment (rbuisson) [inline src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:123]: `filestore`

Comment (rbuisson) [reply src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:123]: What does this service backs up?

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+          volumes:
+          - name: {{ .Release.Name }}-backup-services-storage
+            persistentVolumeClaim:
+              claimName: {{ .Release.Name }}-backup-services-pvc
+---
+
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Release.Name }}-openelis-backup
```
Comment (rbuisson) [inline src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:206]: `{{ .Release.Name }}-openelis-db-backup`

**Review by rbuisson — CHANGES_REQUESTED** [review]
Thanks. Few comments.

File: src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ .Release.Name }}-openelis-db-config
+                  key: OPENELIS_DB_PASSWORD
+            imagePullPolicy: IfNotPresent
+
+            volumeMounts:
+            - mountPath: "/opt/backup"
+              name: {{ .Release.Name }}-backup-services-storage
+              subPath: "postres"
```
[the remark this answers is not in the document]
Comment (rbuisson) [reply src/bahmni-helm/templates/apps/backup-services/cron-jobs.yml:244]: post**g**res

**Review by rbuisson — CHANGES_REQUESTED** [review]
typo

**Comment by enyachoke** [comment]
_typo: postres -> postgres_  I renamed the subpath as well.

**Review by rbuisson — APPROVED** [review]

**Pull request merged**